ogr/ogrsf_frmts/hana/ogrhanadatasource.cpp

```diff
@@ -695,6 +696,25 @@ void OGRHanaDataSource::CreateTable(
     ExecuteSQL(sql);
 }
 
+void OGRHanaDataSource::UpdateCloudVersion()
+{
+    if (hanaVersion_.major() < 4)
+    {
+        cloudVersion_ = HANAVersion(0, 0, 0);
+        return;
+    }
+
+    odbc::StatementRef stmt = conn_->createStatement();
+    const char *sql = "SELECT CLOUD_VERSION FROM SYS.M_DATABASE;";
+
+    odbc::ResultSetRef rsVersion = stmt->executeQuery(sql);
+    if (rsVersion->next())
```

Comment:
What is the value of cloudVersion_ if rsVersion->next() returns false?

Reply:
I adjusted the default constructor to make this explicit: In this case the version is 0.0.0